include/caffe/filler.hpp

```diff
@@ -60,6 +61,43 @@ class UniformFiller : public Filler<Dtype> {
   }
 };
 
+/// @brief Fills a Blob with uniformly distributed values @f$ x\sim U(a, b) @f$.
+//         RNG is only called once, and its output is stored in
+//         UniformStaticFillerdata. Followup calls to the Filler function
+//         will bypass RNG.
+template <typename Dtype>
+class UniformStaticFiller : public Filler<Dtype> {
+ public:
+  explicit UniformStaticFiller(const FillerParameter& param)
+      : Filler<Dtype>(param) {
+        UniformStaticFillerFlag = false;
+      }
+  virtual void Fill(Blob<Dtype>* blob) {
+    Dtype* data = blob->mutable_cpu_data();
+    const int blob_count = blob->count();
+    CHECK(blob_count);
+
+    if (UniformStaticFillerFlag == false) {
+      UniformStaticFillerdata = reinterpret_cast<Dtype*>(
+          malloc(blob_count*sizeof(Dtype)));
+      caffe_rng_uniform(blob_count, Dtype(this->filler_param_.min()),
+          Dtype(this->filler_param_.max()), UniformStaticFillerdata);
+      UniformStaticFillerFlag = true;
+    }
+    CHECK_EQ(this->filler_param_.sparse(), -1)
+        << "Sparsity not supported by this Filler.";
+    caffe_copy(blob_count, UniformStaticFillerdata, data);
+  }
+  virtual ~UniformStaticFiller() {
+    free(UniformStaticFillerdata);
+  }
+
+ private:
+  Dtype* UniformStaticFillerdata;
+  bool UniformStaticFillerFlag;
```

Comment:
Try this:
`std::vector<Dtype> UniformStaticFillerdata;` 
Then on line 81 call:
`UniformStaticFillerdata.resize(blob_count);`